src/snowflake/snowpark/_internal/ast.py

```diff
@@ -20,14 +25,27 @@ def expr_to_dataframe_expr(expr):
     return dfe
 
 
+def ast_expr_from_python_val(expr, val):
+    if isscalar(val):
+        expr.scalar = val
+    elif isinstance(val, DataFrame):
+        expr.dataframe_expr = val
+    elif isinstance(val, Series):
+        expr.series_expr = val
+    elif isinstance(val, slice):
```

Comment:
slice and lists should go before dataframe and series, since they're Python values.